src/services/file/service.ts

```diff
@@ -96,21 +104,25 @@ class FileService {
       });
     }
 
-    return this.repository.getUrl(
-      {
-        expiration,
-        filepath,
-        id,
-      },
-      this.logger,
-    );
+    const getUrl = () =>
+      this.repository.getUrl(
+        {
+          expiration,
+          filepath,
+          id,
+        },
+        this.logger,
+      );
```

Comment:
Could this be in a private function ?

Reply:
I can put it into a private function, but there is no advantage compared to store it in a const.

```
private async t({
    expiration,
    filepath,
    id,
  }: {
    expiration?: number;
    id?: string;
    filepath: string;
  }) {
    return this.repository.getUrl(
      {
        expiration,
        filepath,
        id,
      },
      this.logger,
    );
  }

  async getUrl(data: { expiration?: number; id?: string; path?: string }): Promise<string> {
    const { expiration, id, path: filepath } = data;
    if (!filepath || !id) {
      throw new DownloadFileInvalidParameterError({
        filepath,
        id,
      });
    }

    return (
      this.caching?.getOrCache(filepath, () => this.t({ expiration, filepath, id }), expiration) ??
      this.t({ expiration, filepath, id })
    );
  }
```

What do you think @Defferrard?

Reply:
It is mainly to improve the readability. However, I'm okay with both solutions